test/unit/foreman/renderer_test.rb

```diff
@@ -97,6 +97,13 @@ def setup_safemode_renderer
       end
     end
 
+    test "#{renderer_name} should raise renderer syntax error on syntax error" do
+      send "setup_#{renderer_name}"
+      assert_raises Foreman::Renderer::SyntaxError do
```

Comment:
Can you add assertions to check the exception's correct? (`e = assert_raises...`)

Particularly that the backtrace contains the template name and the message has correctly wrapped the original syntax error.